cofdb_submit/main.py

```diff
@@ -234,32 +197,45 @@ def servable(self):
             self.btn_add_cif,
         )
         return self.column.servable()
-        
+
     def on_click_parse(self, event):
+        """Load the CIF, unwrap it to P1 using ASE, extract some info, and display it.
+        NOTE: it would be nice to use the filename as CIF name, but FileInput doesn't allow you to know the path.
+        """
         from ase.io import read, write
         from ase.io.cif import write_cif
         from ase.geometry.dimensionality import analyze_dimensionality
         from ase.build import make_supercell
         from io import StringIO, BytesIO
         import re
-        
+
+        # turn "Add CIF" button primary, to remember clicking it again!
+        self.btn_add_cif.button_type = 'primary'
+
+        self.inp_name.value = self.inp_cif.filename.split(".")[0]
+
         cif_str = self.inp_cif.value.decode()
         atoms = read(StringIO(cif_str), format='cif')
-        
+
         formula = atoms.get_chemical_formula()
         elements = [e for e in re.split(r'\d+', formula) if e]
         self.inp_elements.value = ",".join(elements)
-        
+        self.inp_modifications.value = 'none'
+
         intervals = analyze_dimensionality(atoms, method='RDA')
-        if intervals[0].dimtype == '2D' or True:
+        if intervals[0].dimtype == '2D':
             self.inp_dimensionality.value = '2D'
-            
-            # multiply cell if cell vectors are too short
+
+            # Check if it is correcly oriented, and extend to two layers if onyly one is present
+            z_min_thr = 6 #if less, it is likely a single layer
             cell_lengths = atoms.get_cell_lengths_and_angles()[0:3]
-            if any(cell_lengths < 4.5):
+            if min(cell_lengths[0],cell_lengths[1] < z_min_thr): # X or Y are perpendicular to a single layer
```

Comment:
not sure this is what you want 
```
>>> 0,1 < 1
(0, False)
>>> min(0,1 < 1)
0
```
Also, it's not very readable. Let's go for this
```suggestion
            if cell_lengths[0] < z_min_thr or cell_lengths[1] < z_min_thr: # X or Y are perpendicular to a single layer
```

Reply:
Right, I placed the bracket in the wrong position. 
I'm going to correct it and also comment the code a little more.